fendermint/vm/interpreter/src/fvm/genesis/builtin.rs

```diff
@@ -0,0 +1,246 @@
+
+        // Cron actor
+        state
+            .create_builtin_actor(
+                cron::CRON_ACTOR_CODE_ID,
+                cron::CRON_ACTOR_ID,
+                &cron::State {
+                    entries: vec![], // TODO: Maybe with the IPC.
+                },
+                TokenAmount::zero(),
+                None,
+            )
+            .context("failed to create cron actor")?;
+
+        // Burnt funds actor (it's just an account).
+        state
+            .create_builtin_actor(
+                account::ACCOUNT_ACTOR_CODE_ID,
+                burntfunds::BURNT_FUNDS_ACTOR_ID,
+                &account::State {
+                    address: burntfunds::BURNT_FUNDS_ACTOR_ADDR,
+                },
+                TokenAmount::zero(),
+                None,
+            )
+            .context("failed to create burnt funds actor")?;
+
+        // A placeholder for the reward actor, beause I don't think
+        // using the one in the builtin actors library would be appropriate.
+        // This effectively burns the miner rewards. Better than panicking.
+        state
+            .create_builtin_actor(
+                account::ACCOUNT_ACTOR_CODE_ID,
+                reward::REWARD_ACTOR_ID,
+                &account::State {
+                    address: reward::REWARD_ACTOR_ADDR,
+                },
+                TokenAmount::zero(),
+                None,
+            )
+            .context("failed to create reward actor")?;
+
+        let eam_state = fendermint_actor_eam::State::new(
+            &state.store(),
+            genesis
+                .contract_deployers
+                .clone()
+                .into_iter()
+                .map(|a| a.0)
+                .collect(),
+        )?;
+        state
+            .create_actor_internal(
+                eam_code_cid,
+                eam::EAM_ACTOR_ID,
+                &eam_state,
+                TokenAmount::zero(),
+                None,
+            )
+            .context("failed to create EAM actor")?;
+
+        Ok(addr_to_id)
+    }
+
+    async fn replace_eam_actor<DB>(
+        &self,
+        state: &mut FvmGenesisState<DB>,
+        store: &MemoryBlockstore,
+        builtin: &mut Vec<CodeByName>,
+    ) -> anyhow::Result<(Cid, Cid)>
+    where
+        DB: Blockstore + 'static + Send + Sync + Clone,
+    {
+        let (custom_manifest_version, custom_manifest_data_cid): (u32, Cid) =
+            parse_bundle(&store, &state.custom_actor_bundle).await?;
+        let custom_actor_manifest =
+            CustomActorManifest::load(&store, &custom_manifest_data_cid, custom_manifest_version)?;
+
+        let code = custom_actor_manifest
+            .code_by_name(fendermint_actor_eam::IPC_EAM_ACTOR_NAME)
+            .ok_or_else(|| anyhow!("ipc eam actor not found"))?;
+
+        for (_, code_cid) in builtin.iter_mut().filter(|(name, _)| name == "eam") {
+            *code_cid = *code
+        }
```

Comment:
do we not have a constant for the name?, also we should assert that this name was actually found in the builtin vector

Reply:
We added one in https://github.com/consensus-shipyard/ipc/pull/610/files#diff-a5d767cb02a5b7648a64451054b2e648413395275d693e7ea683dd9676d01115R129 , although not for this and it's not accessible, but we could move that type to a common place.

The FVM has them [here](https://github.com/filecoin-project/ref-fvm/blob/d3913e483b9234cadf661e5fd9e40632287782ea/fvm/src/machine/manifest.rs#L10-L15) but they aren't accessible.